### DISCUSSION TITLE
Anyway to receive mails using IMAP in deno.

### DISCUSSION THREAD

**Discussion by shivajivarma** [denoland/deno#22526] [Q&A]
`npm:imap@0.8.19` still does not work. Not sure we have any other lib to achieve it.

```js
import Imap from 'npm:imap@0.8.19';

const imapConfig: Imap.Config = {
    "user": "############",
    "password": "###########",
    "host": "outlook.office365.com",
    "port": 993,
    "tls": true,
};

const imap = new Imap(imapConfig);

imap.once('ready', () => {
    console.info('ready');
});

imap.once('error', (err: Error) => {
    console.error(err);
});

imap.once('end', () => {
    console.info('Connection ended');
});

imap.connect();
```

This still throws error
![Screenshot 2024-02-22 123656](https://github.com/denoland/deno/assets/1388742/ccd8f720-5dd1-4bac-8073-5bedb8226e6e)

**Comment by nnmrts** [comment]
All I know is that several `.rid` references got removed/deprecated in the last few releases. Maybe this is a bug in the npm resolver or maybe you just need to delete your deno.lock and/or node_modules folder.

**Comment by CetinSert** [comment]
`imap-simple` also does not work (might even depend on `imap`)
